eslint-custom-rules/nbsp-in-french-translations.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * This rule aims to spot miss-use of a space instead of a non-breaking space (nbsp) with code \u00a0
+ * before some characters in French translations, such as ! ? : ; € » or after «
+ *
+ * Ex: "Bienvenue !" should be "Bienvenue\u00a0!"
+ */
+
+module.exports = {
+  name: 'nbsp-in-french-translations',
+  meta: {
+    docs: {
+      description:
+        'force the use of non-breaking space before some characters in French translations',
+    },
+    fixable: 'code',
+  },
+  create(context) {
+    return {
+      'TaggedTemplateExpression[tag.name="t"] > TemplateLiteral > TemplateElement[value.raw=/ !/]':
+        (node) => {
+          context.report({
+            node,
+            message:
+              'Please use \\u00a0 (nbsp) instead of whitespace before !, ?, :, ;, €, » and after «',
+            fix: function (fixer) {
+              const textToReplace = node.value.raw.replaceAll(/ !/g, '\\u00a0!')
+              return fixer.replaceText(node, `\`${textToReplace}\``)
+            },
+          })
+        },
+      'TaggedTemplateExpression[tag.name="t"] > TemplateLiteral > TemplateElement[value.raw=/ \\?/]':
+        (node) => {
+          context.report({
+            node,
+            message:
+              'Please use \\u00a0 (nbsp) instead of whitespace before !, ?, :, ;, €, » and after «',
+            fix: function (fixer) {
+              const textToReplace = node.value.raw.replaceAll(/ \?/g, '\\u00a0?')
```

Comment:
j'ai l'impression que le message est plus complet que la règle

serait-il possible de compléter avec quelque chose du genre ?

```suggestion
              const textToReplace = node.value.raw.replaceAll(/  ([!?:;€»])/g, '\\u00a0$1')
```

---

j'imagine qu'on devrait avoir 2 règles : 
1. celle qui vérifie qu'il y a un nbsp après certains caractères
2. celle qui vérifie qu'il y a un nbsp avant certains caractères

Reply:
Je comptais le faire progressivement (à chaque fois que je remplace qqch, j'ai des tests à fix), c'est pour ça 

Mais bon retour pour capturer le caractère dans la regex, au lieu de dupliquer comme un cochon comme j'ai fait 😅